packages/flutter/lib/src/rendering/image_filter_config.dart

```diff
@@ -0,0 +1,258 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// @docImport 'package:flutter/widgets.dart';
+library;
+
+import 'dart:ui' as ui;
+
+import 'package:flutter/foundation.dart';
+
+/// Context information provided when resolving an [ImageFilterConfig].
+///
+///  See also:
+///  * [ImageFilterConfig.resolve], which takes an instance of this class to
+///    create a [ui.ImageFilter].
+@immutable
+class ImageFilterContext {
+  /// Creates an [ImageFilterContext].
+  const ImageFilterContext({required this.bounds});
+
+  /// The bounds to apply the filter in the local coordinate space.
+  ///
+  /// Specified in the canvas's current coordinate space and affected by the
+  /// current transform. The bounds may not be axis-aligned in final canvas
+  /// coordinates.
+  ///
+  /// Typically the bounds of the widget or render object applying the filter.
+  final ui.Rect bounds;
+}
+
+/// A description of an image filter that can adapt to the layout of its target.
+///
+/// Use [ImageFilterConfig] to define visual effects that depend on the size,
+/// position, or other layout attributes of a widget or render object.
+///
+/// Unlike [ui.ImageFilter], which is an engine-level object with static
+/// parameters, [ImageFilterConfig] acts as a framework-level blueprint. It
+/// delays the creation of the actual filter until the painting phase, where
+/// it is resolved into a [ui.ImageFilter] using an [ImageFilterContext].
+///
+/// This resolution process allows filters to use layout information as
+/// parameters. For example, a filter can use the [Rect] bounds provided by
+/// the context to restrict its sampling area to the object's boundaries.
+///
+/// Most layout-independent filters can be wrapped using the default
+/// constructor. For effects that require layout information, such as a
+/// "bounded" blur, use the specialized constructors.
+///
+/// See also:
+///
+///  * [ImageFilterContext], which provides the layout information used to
+///    resolve this configuration.
+///  * [ImageFilterConfig.blur], which can be used to create a "bounded blur"
+///    that limits sampling to the object's boundaries.
+///  * [ui.ImageFilter], the engine-level class that this config resolves to.
+///  * [BackdropFilter.filterConfig], which uses this class to configure its effect.
+@immutable
+abstract class ImageFilterConfig {
+  /// Creates a configuration that directly uses the given filter.
```

Comment:
Maybe mention this won't resolve the given image filter with the rect if they want to use ImageFilterConfig(ImageFilter.blur), they should use ImageFilterConfig.blur